plugins/nf-tower/src/main/io/seqera/tower/plugin/cli/AuthCommandImpl.groovy

```diff
@@ -0,0 +1,1354 @@
+        tableLines.each { println it }
+    }
+
+    private List<String> generateStatusTableLines(List<List<String>> rows) {
+        if( !rows ) return []
+
+        final List<String> lines = []
+
+        // Calculate column widths (accounting for ANSI codes)
+        def col1Width = rows.collect { stripAnsiCodes(it[0]).length() }.max()
+        def col2Width = rows.collect { stripAnsiCodes(it[1]).length() }.max()
+        def col3Width = rows.collect { stripAnsiCodes(it[2]).length() }.max()
+
+        // Add some padding
+        col1Width = Math.max(col1Width, 15) + 2
+        col2Width = Math.max(col2Width, 15) + 2
+        col3Width = Math.max(col3Width, 10) + 2
+
+        // Add table header
+        lines.add(ColorUtil.colorize("${'Setting'.padRight(col1Width)} ${'Value'.padRight(col2Width)} Source", "cyan bold"))
+        lines.add("${'-' * col1Width} ${'-' * col2Width} ${'-' * col3Width}".toString())
+
+        // Add rows
+        rows.each { row ->
+            final paddedCol1 = padStringWithAnsi(row[0], col1Width)
+            final paddedCol2 = padStringWithAnsi(row[1], col2Width)
+            final paddedCol3 = ColorUtil.colorize(row[2], 'dim', true)
+            lines.add("${paddedCol1} ${paddedCol2} ${paddedCol3}".toString())
+        }
+
+        return lines
+    }
+
+    private String stripAnsiCodes(String text) {
+        return text?.replaceAll(/\u001B\[[0-9;]*m/, '') ?: ''
+    }
+
+    private String padStringWithAnsi(String text, int width) {
+        final plainText = stripAnsiCodes(text)
+        final padding = width - plainText.length()
+        return padding > 0 ? text + (' ' * padding) : text
+    }
+
+    private String shortenPath(String path) {
+        final userHome = System.getProperty('user.home')
+        if( path.startsWith(userHome) ) {
+            return '~' + path.substring(userHome.length())
+        }
+        return path
+    }
+
+    private Map getConfigValue(Map config, Map auth, String configKey, String envVarName, String defaultValue = null) {
+        //Checks where the config value came from
+        final authValue = auth[configKey]
+        final configValue = config[configKey]
+        final envValue = envVarName ? SysEnv.get(envVarName) : null
+        final effectiveValue = authValue ?: configValue ?: envValue ?: defaultValue
+
+        def source = null
+        if( authValue ) {
+            source = shortenPath(getAuthFile().toString())
+        } else if( configValue ) {
+            source = shortenPath(getConfigFile().toString())
+        } else if( envValue ) {
+            source = "env var \$${envVarName}"
+        } else if( defaultValue ) {
+            source = "default"
+        }
+
+        return [
+            value     : effectiveValue,
+            source    : source,
+            fromConfig: configValue != null || authValue != null,
+            fromEnv   : envValue != null,
+            isDefault : !authValue && !configValue && !envValue
+        ]
+    }
```

Comment:
This looks a bit magic. Is there a solid unit test for it ?

Reply:
Yeah this function grew a bit over time. The tests are pretty extensive though:

https://github.com/nextflow-io/nextflow/blob/7c0c8e43209805a2240baa38f5b8928c878629cf/plugins/nf-tower/src/test/io/seqera/tower/plugin/cli/AuthCommandImplTest.groovy#L354-L496